dvc/external_repo.py

```diff
@@ -65,66 +68,206 @@ def clean_repos():
         _remove(path)
 
 
-class ExternalRepo(Repo):
-    def __init__(self, root_dir, url):
-        super().__init__(root_dir)
-        self.url = url
-        self._set_cache_dir()
-        self._fix_upstream()
+class BaseExternalRepo:
+    _tree_rev = None
+    _local_cache = None
+
+    @contextmanager
+    def use_cache(self, cache):
```

Comment:
This context manager lets us use an existing local cache for downloading instead of the default tmpdir erepo cache